pinot-common/src/test/java/org/apache/pinot/common/metrics/PinotJMXToPromMetricsTest.java

```diff
@@ -0,0 +1,266 @@
+    Assert.assertTrue(promMetrics.contains(PromMetric.withNameAndLabels(
+        exportedMetricPrefix + exportedGaugePrefix + "_" + "Value", labels)));
+  }
+
+  protected void assertTimerExportedCorrectly(String exportedTimerPrefix, String exportedMetricPrefix) {
+    List<PromMetric> promMetrics = null;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+    for (String meterType : TIMER_TYPES) {
+      Assert.assertTrue(promMetrics.contains(PromMetric.withName(
+          exportedMetricPrefix + exportedTimerPrefix + "_" + meterType)));
+    }
+  }
+
+  protected void assertTimerExportedCorrectly(String exportedTimerPrefix, List<String> labels,
+      String exportedMetricPrefix) {
+    List<PromMetric> promMetrics;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+    for (String meterType : METER_TYPES) {
+      Assert.assertTrue(promMetrics.contains(PromMetric.withNameAndLabels(
+          exportedMetricPrefix + exportedTimerPrefix + "_" + meterType, labels)));
+    }
+  }
+
+  protected void assertMeterExportedCorrectly(String exportedMeterPrefix, String exportedMetricPrefix) {
+    List<PromMetric> promMetrics = null;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+    for (String meterType : METER_TYPES) {
+      Assert.assertTrue(promMetrics.contains(
+              PromMetric.withName(exportedMetricPrefix + exportedMeterPrefix + "_" + meterType)),
+          exportedMeterPrefix);
+    }
+  }
+
+  protected void assertMeterExportedCorrectly(String exportedMeterPrefix, List<String> labels,
+      String exportedMetricPrefix) {
+    List<PromMetric> promMetrics = null;
+    try {
+      promMetrics = parseExportedPromMetrics(getExportedPromMetrics().getResponse());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e);
+    }
+    for (String meterType : METER_TYPES) {
+      Assert.assertTrue(promMetrics.contains(PromMetric.withNameAndLabels(
+          exportedMetricPrefix + exportedMeterPrefix + "_" + meterType, labels)), exportedMeterPrefix);
+    }
+  }
+
+  protected List<PromMetric> parseExportedPromMetrics(String response)
+      throws IOException {
+
+    List<PromMetric> exportedPromMetrics = new ArrayList<>();
+
+    BufferedReader reader = new BufferedReader(new StringReader(response));
```

Comment:
Please wrap this into a try with resources to be sure it is closed even if there is a failure. Given it is a test that is not very important, but if in the future we add error prone plugin, this will be detected as an issue.

Reply:
Done